src/Node/ReadLine.purs

```diff
@@ -1,42 +1,92 @@
 -- | This module provides a binding to the Node `readline` API.
 
-module Node.ReadLine where
+module Node.ReadLine
+  ( Interface
+  , READLINE
+  , InterfaceOptions
+  , Completer
+  , LineHandler
+  , createInterface
+  , createConsoleInterface
+  , noCompletion
+  , prompt
+  , setPrompt
+  , setLineHandler
+  , close
+  ) where
 
 import Prelude (return)
-
-import Control.Monad.Eff
-import Control.Monad.Eff.Console
+import Control.Monad.Eff (Eff)
+import Control.Monad.Eff.Console (CONSOLE)
+import Control.Monad.Eff.Exception (EXCEPTION)
+import Data.Maybe (Maybe(Just, Nothing), fromMaybe)
+import Node.Process (stdin, stdout)
+import Node.Stream (Readable, Writable)
 
 -- | A handle to a console interface.
 -- |
 -- | A handle can be created with the `createInterface` function.
 foreign import data Interface :: *
 
+-- | The effect of interacting with a stream via an `Interface`
+foreign import data READLINE :: !
+
+foreign import createInterfaceImpl :: forall opts eff. {| opts} -> Eff (readline :: READLINE | eff) Interface
```

Comment:
But we would have to expose those newtypes in order for people to be able to use them, right?

I wonder if we might be able to create a version of `purescript-options` without the `IsOption` type class. The only instance of `IsOptions` that does anything different from what is conceptually the same as constructing a `Tuple String Foreign` value is the `Maybe` instance. Additionally, it might be the case that not all `Maybe` values want to be turned into `Options` values in this exact way - really, it depends on the specific context, right? Maybe the answer is that the `Option` should contain the `assoc` function?

```purescript
type Option opt value =
  { key :: String
  , toOptions :: value -> Options opt
  }
```